### PR TITLE
Existence Check Added to Get requests

### PR DESCRIPTION
**Pull request by istairbn**
getAllVersions and getLatestVersion tasks were completing in XL Release, even if the CI didn't exist. That meant downstream tasks were failing because an empty string was being passed to them. 

The pre-existing workaround was to do a separate doesCIExist task for every single Get, which seemed a lot of work considering we just want to catch mistakes. 

Added in a second check to ensure that the Application has at least one child package. This means that tese two tasks should now always return something or fail, rather than sometimes returning an empty string.

### CODE REVIEW

**Comment by jdewinne** [comment]
How about making this configurable? So, similar to this [property](https://github.com/xebialabs-community/xlr-xldeploy-plugin/blob/master/src/main/resources/synthetic.xml#L108)

**Comment by istairbn** [comment]
Valid point - no sense in breaking anyone's workflow if they are using the empty return as a conditional.
Added a throwOnFail parameter.

File: src/main/resources/xldeploy/getAllVersionsTask.py [hunk 1]
```diff
@@ -6,7 +6,20 @@
 
 from xldeploy.XLDeployClientUtil import XLDeployClientUtil
 
-
 xld_client = XLDeployClientUtil.create_xldeploy_client(xldeployServer, username, password)
+try:
```
Comment (jdewinne) [inline src/main/resources/xldeploy/getAllVersionsTask.py]: Why not make throwOnFail having a default `false` value in the `synthetic.xml`

File: src/main/resources/xldeploy/getLatestVersionTask.py [hunk 2]
```diff
@@ -6,10 +6,25 @@
 
 from xldeploy.XLDeployClientUtil import XLDeployClientUtil
 
-
 xld_client = XLDeployClientUtil.create_xldeploy_client(xldeployServer, username, password)
 
+try:
+	throwOnFail
+except:
+	throwOnFail = False
```
Comment (jdewinne) [inline src/main/resources/xldeploy/getLatestVersionTask.py]: Why not make throwOnFail having a default `false` value in the `synthetic.xml`

**Comment by istairbn** [comment]
Good point @jdewinne - I've made the changes on your suggestion.

**Pull request merged**